lib/src/model.dart

```diff
@@ -1487,47 +1614,85 @@ abstract class ModelElement implements Comparable, Nameable, Documentable {
   // somehow.
   ModelElement(this._element, this._library);
 
-  factory ModelElement.from(Element e, Library library) {
-    if (e.kind == ElementKind.DYNAMIC) {
-      return new Dynamic(e, library);
-    }
-    // Also handles enums
-    if (e is ClassElement) {
-      return new Class(e, library);
-    }
-    if (e is FunctionElement) {
-      return new ModelFunction(e, library);
-    }
-    if (e is FunctionTypeAliasElement) {
-      return new Typedef(e, library);
-    }
-    if (e is FieldElement) {
-      return new Field(e, library);
-    }
-    if (e is ConstructorElement) {
-      return new Constructor(e, library);
-    }
-    if (e is MethodElement && e.isOperator) {
-      return new Operator(e, library);
-    }
-    if (e is MethodElement && !e.isOperator) {
-      return new Method(e, library);
-    }
-    if (e is TopLevelVariableElement) {
-      return new TopLevelVariable(e, library);
-    }
-    if (e is PropertyAccessorElement) {
-      return new Accessor(e, library);
-    }
-    if (e is TypeParameterElement) {
-      return new TypeParameter(e, library);
-    }
-    if (e is ParameterElement) {
-      return new Parameter(e, library);
+  /// Do not construct any ModelElements unless they are from this constructor.
+  /// TODO(jcollins-g): enforce this.
+  /// Specify enclosingClass only if this is to be an inherited object.
+  /// TODO(jcollins-g): this way of using the optional parameter is messy,
+  /// clean that up.
+  factory ModelElement.from(Element e, Library library,
+      {Class enclosingClass}) {
+    Tuple3<Element, Library, Class> key =
```

Comment:
Yes, this is what was needed. Knew that dartdoc was likely creating ModelElements for the same element multiple times, but this was just a todo, until now!